src/d2d1.rs

```diff
@@ -0,0 +1,1405 @@
+//-------------------------------------------------------------------------------------------------
+/// Allows additional parameters for factory creation.
+#[repr(C)]
+#[derive(Clone, Copy, Debug)]
+pub struct D2D1_FACTORY_OPTIONS {
+    pub debugLevel: D2D1_DEBUG_LEVEL,
+}
+
+//-------------------------------------------------------------------------------------------------
+// ID2D1Resource
+//-------------------------------------------------------------------------------------------------
+RIDL!(
+interface ID2D1Resource(ID2D1ResourceVtbl): IUnknown(IUnknownVtbl) {
+    fn GetFactory(&mut self, factory: *mut *mut ID2D1Factory) -> ::HRESULT
+});
+
+//-------------------------------------------------------------------------------------------------
+// ID2D1Image
+//-------------------------------------------------------------------------------------------------
+RIDL!(
+interface ID2D1Image(ID2D1ImageVtbl): ID2D1Resource(ID2D1ResourceVtbl) {
+});
+
+//-------------------------------------------------------------------------------------------------
+// ID2D1Bitmap
+//-------------------------------------------------------------------------------------------------
+RIDL!(
+interface ID2D1Bitmap(ID2D1BitmapVtbl): ID2D1Image(ID2D1ImageVtbl) {
+    fn GetSize(&mut self) -> D2D1_SIZE_F,
+    fn GetPixelSize(&mut self) -> D2D1_SIZE_U,
+    fn GetPixelFormat(&mut self) -> D2D1_PIXEL_FORMAT,
+    fn GetDpi(&mut self, dpiX: *mut ::FLOAT, dpiY: *mut ::FLOAT) -> (),
+    fn CopyFromBitmap(
+        &mut self, destPoint: *const D2D1_POINT_2U, bitmap: *mut ID2D1Bitmap,
+        srcRect: *const D2D1_RECT_U
+    ) -> ::HRESULT,
+    fn CopyFromRenderTarget(
+        &mut self, destPoint: *const D2D1_POINT_2U, renderTarget: *mut ID2D1RenderTarget,
+        srcRect: *const D2D1_RECT_U
+    ) -> ::HRESULT,
+    fn CopyFromMemory(
+        &mut self, dstRect: *const D2D1_RECT_U, srcData: *const ::c_void, pitch: ::UINT32
+    ) -> ::HRESULT
+});
+
+//-------------------------------------------------------------------------------------------------
+// ID2D1GradientStopCollection
+//-------------------------------------------------------------------------------------------------
+RIDL!(
+interface ID2D1GradientStopCollection(ID2D1GradientStopCollectionVtbl)
+    : ID2D1Resource(ID2D1ResourceVtbl) {
+    fn GetGradientStopCount(&mut self) -> ::UINT32,
+    fn GetGradientStops(
+        &mut self, gradientStops: *mut D2D1_GRADIENT_STOP, gradientStopsCount: ::UINT32
+    ) -> (),
+    fn GetColorInterpolationGamma(&mut self) -> D2D1_GAMMA,
+    fn GetExtendMode(&mut self) -> D2D1_EXTEND_MODE
+});
+
+//-------------------------------------------------------------------------------------------------
+// ID2D1Brush
+//-------------------------------------------------------------------------------------------------
+RIDL!(
+interface ID2D1Brush(ID2D1BrushVtbl): ID2D1Resource(ID2D1ResourceVtbl) {
+    fn SetOpacity(&mut self, opacity: ::FLOAT) -> (),

```

Comment:
It feels weird seeing `-> ()`, but then I remember that `RIDL!` requires a return type be specified.